src/validator.cpp

```diff
@@ -949,6 +964,50 @@ void Validator::ValidatorImpl::validateMathMLElements(const XmlNodePtr &node, co
     }
 }
 
+bool Validator::ValidatorImpl::validateVariableInterface(const VariablePtr &variable, Variable::InterfaceType interfaceType)
+{
+    auto result = interfaceTypeToString.find(interfaceType);
+    const std::string interfaceTypeString = result->second;
+    return variable->interfaceType().find(interfaceTypeString) != std::string::npos;
+}
+
+void Validator::ValidatorImpl::validateEquivalenceInterface(const VariablePtr &variable1, const VariablePtr &variable2)
+{
+    InterfaceTypePair pair = determineInterfaceType(variable1, variable2);
+    if (pair.first == Variable::InterfaceType::NONE && pair.second == Variable::InterfaceType::NONE) {
+        std::string component1Name;
+        std::string component2Name;
+        auto component1 = std::dynamic_pointer_cast<Component>(variable1->parent());
+        auto component2 = std::dynamic_pointer_cast<Component>(variable2->parent());
+        if (component1 != nullptr) {
+            component1Name = component1->name();
+        }
+        if (component2 != nullptr) {
+            component2Name = component2->name();
+        }
+        ErrorPtr err = Error::create();
+        err->setDescription("Invalid equivalence between '" + variable1->name() + "' and '" + variable2->name() + "', the separation of components '" + component1Name + "' and '" + component2Name + "' is too great.");
+        err->setVariable(variable1);
+        err->setKind(Error::Kind::CONNECTION);
+        mValidator->addError(err);
+    } else {
+        if (!validateVariableInterface(variable1, pair.first)) {
+            ErrorPtr err = Error::create();
```

Comment:
Can we add a "I expected (something) or (something) but I found (something else)" end to these statements? Makes it more useful to the user

Reply:
I think we have an issue for going over all descriptions don't we?